src/H5Iint.c

```diff
@@ -1069,7 +1069,8 @@ H5I_dec_ref(hid_t id)
     FUNC_ENTER_NOAPI((-1))
 
     /* Sanity check */
-    HDassert(id >= 0);
+    if (id < 0)
```

Comment:
Since this is library-internal, I think the assert is fine and there shouldn't be a need to check for id < 0. That's the point of the assertion in debug builds.

Reply:
Corrected.